src/Cryptol/Eval/Value.hs

```diff
@@ -375,9 +392,18 @@ mkSeq :: Backend sym => sym -> Nat' -> TValue -> SeqMap sym (GenValue sym) -> SE
 mkSeq sym len elty vals = case len of
   Nat n
     | isTBit elty -> VWord <$> bitmapWordVal sym n (fromVBit <$> vals)
-    | otherwise   -> pure $ VSeq n vals
+    | otherwise   -> pure $ VSeqCtor n vals
   Inf             -> pure $ VStream vals
 
+mkFinSeq :: Backend sym => sym -> Integer -> [GenValue sym] -> GenValue sym
```

Comment:
Please add some Haddocks for this function. What does it construct, and where would it be useful to use this function? (I _think_ `mkFinSeq sym n vs` meant to abstract over the pattern of writing `VSeq n (finiteSeqMap sym vs)`, but it would be helpful to say as much.)